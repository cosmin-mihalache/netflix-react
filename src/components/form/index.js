import React from 'react';
import {
  Container,
  Base,
  Title,
  Submit,
  Text,
  SmallText,
  Input,
  Link,
  Error,
} from './styles/form';

export default function Form({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

Form.Base = function FormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

Form.Title = function FormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Form.Error = function FormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

Form.Input = function FormInput({ children, ...restProps }) {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = function FormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};

Form.Link = function FormLink({ children, ...restProps }) {
  return <Link {...restProps}>{children}</Link>;
};

Form.Text = function FormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Form.SmallText = function FormSmallText({ children, ...restProps }) {
  return <SmallText {...restProps}>{children}</SmallText>;
};
