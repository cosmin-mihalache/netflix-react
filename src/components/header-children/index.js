import React from 'react';
import { Container, Title, SubTitle } from './styles/header-children';

export default function HeaderChild({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

HeaderChild.Title = function HeaderChildTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

HeaderChild.SubTitle = function HeaderChildSubTitle({
  children,
  ...restProps
}) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};
