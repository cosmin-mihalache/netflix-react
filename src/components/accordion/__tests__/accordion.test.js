import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import dataFaq from '../../../fixtures/faq.json';
import { Accordion } from '../../../components';

describe('Accordion', () => {
  it('renders the <Accordion /> with populated data', () => {
    const { getByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>

        {dataFaq.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>{item.body}</Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>
    );

    expect(getByText('Frequently Asked Questions')).toBeTruthy();
    expect(getByText('What is Netflix?')).toBeTruthy();
    expect(getByText('How much does Netflix cost?')).toBeTruthy();
    expect(getByText('Where can I watch?')).toBeTruthy();
    expect(getByText('How do I cancel?')).toBeTruthy();
    expect(getByText('What can I watch on Neflix')).toBeTruthy();
  });

  it('opens and closes the <Accordion /> component', () => {
    const { queryByText } = render(
      <Accordion>
        <Accordion.Title>Frequently Asked Questions</Accordion.Title>

        {dataFaq.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>{item.body}</Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>
    );

    const whatCanIWatchOnNetflixBodyText =
      'Netflix has an extensive library of feature films, documentaries, TV shows, anime, award-winning Netflix originals, and more. Watch as much as you want, anytime you want.';

    expect(queryByText(whatCanIWatchOnNetflixBodyText)).toBeFalsy();
    fireEvent.click(queryByText('What can I watch on Neflix'));
    expect(queryByText(whatCanIWatchOnNetflixBodyText)).toBeTruthy();

    fireEvent.click(queryByText('What can I watch on Neflix'));
    expect(queryByText(whatCanIWatchOnNetflixBodyText)).toBeFalsy();
  });
});
