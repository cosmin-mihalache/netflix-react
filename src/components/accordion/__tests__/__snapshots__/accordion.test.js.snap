// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-sc-1gyetxg-0 lkWANf"
>
  <div
    class="accordion__Inner-sc-1gyetxg-1 MgMWN"
  >
    <h1
      class="accordion__Title-sc-1gyetxg-3 ijnBSR"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        What can I watch on Neflix
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        Is Netflix good for kids?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Accordion renders the <Accordion /> with populated data 1`] = `
<div
  class="accordion__Container-sc-1gyetxg-0 lkWANf"
>
  <div
    class="accordion__Inner-sc-1gyetxg-1 MgMWN"
  >
    <h1
      class="accordion__Title-sc-1gyetxg-3 ijnBSR"
    >
      Frequently Asked Questions
    </h1>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        What is Netflix?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        How much does Netflix cost?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        Where can I watch?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        How do I cancel?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        What can I watch on Neflix
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-sc-1gyetxg-2 dosBXH"
    >
      <div
        class="accordion__Header-sc-1gyetxg-4 hyulXV"
      >
        Is Netflix good for kids?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
